### PR TITLE
Run `cargo install --debug --locked` on the CI

### LINKED ISSUE #54: `cargo install` problem again
rust-lang-nursery/failure#342

### PR DESCRIPTION
**Pull request by qryxip**
~~Fixes # 54.~~

### CODE REVIEW

**Comment by qryxip** [comment]
By the way, cargo-udeps v0.1.8 is not uploaded to crates.io.

**Comment by est31** [comment]
I'd rather wait for https://github.com/rust-lang-nursery/failure/pull/343 to be merged instead of fixing the quote version. The CI changes look ok though. Can you remove the Cargo.toml changes?

**Comment by est31** [comment]
> By the way, cargo-udeps v0.1.8 is not uploaded to crates.io.

Oh no. Thanks for the pointer, I've published it now manually.

**Comment by est31** [comment]
Also @qryxip the syn dependency can be removed but it still needs a cargo update to get the structopt update in. Could you make one?

**Comment by est31** [comment]
For reference, this was the structopt fix: https://github.com/TeXitoi/structopt/commit/bc43cdfb9252bcfb02e00fd0917ce1aa9dfc0480#diff-f8859fa7599bdffe4bea39369662f6f0

**Comment by qryxip** [comment]
> it still needs a cargo update to get the structopt update in

That's true. I thought the version of `structopt` was still v0.3.7 or earlier.

|                            | `master` = `v0.1.8` | This PR   |
| -------------------------- | ------------------- | --------- |
| `locked`, `dev-deps`       | ✓                  | ✓        |
| `locked`, `avoid-dev-deps` | ✓                  | ~~X~~ ✓  |
| `update`, `avoid-dev-deps` | X                   | ~~✓~~ X  |

**Comment by qryxip** [comment]
I removed the two commit that modify `Cargo.toml` and `Cargo.lock`. But `cargo install --locked` emits a warning (not an error) because `structopt v0.3.9` was yanked.

```console
$ cargo install --path . --locked
  Installing cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)
    Updating crates.io index
warning: package `structopt v0.3.9` in Cargo.lock is yanked in registry `crates.io`, consider running without --locked
    Finished release [optimized] target(s) in 0.57s
   Replacing /home/ryo/.cargo/bin/cargo-udeps
    Replaced package `cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)` with `cargo-udeps v0.1.8 (/home/ryo/src/github.com/est31/cargo-udeps)` (executable `cargo-udeps`)
```

**Comment by qryxip** [comment]
`failure v0.1.7` has just been published.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,19 +55,36 @@ jobs:
...
         uses: actions-rs/cargo@v1
         with:
-          command: test
-          args: --verbose
-        env:
-          CARGO_UDEPS_TEST_TOOLCHAIN: ${{ matrix.toolchain_nightly }}
+          command: install
+          args: --path . --debug --verbose --locked
+
+      # https://github.com/rust-lang-nursery/failure/issues/342
```
Comment (est31) [inline .github/workflows/ci.yml]: Can you remove this comment and activate the section below?

Comment (qryxip) [reply .github/workflows/ci.yml]: Ok. I activated the section.

**Comment by est31** [comment]
Hmm I can't merge, travis seems to be broken.

**Comment by qryxip** [comment]
Passed with this...

```diff
+curl-sys = "=0.4.27"
```

**Comment by est31** [comment]
> +curl-sys = "=0.4.27"

We should debug the issue a little and then file a bug at https://github.com/alexcrichton/curl-rust .

**Pull request merged**